pkg/cli/cmd/radinit/recipe.go

```diff
@@ -28,14 +27,90 @@ import (
 	msg_ctrl "github.com/radius-project/radius/pkg/messagingrp/frontend/controller"
 	recipe_types "github.com/radius-project/radius/pkg/recipes"
 	"github.com/radius-project/radius/pkg/to"
+	"github.com/radius-project/radius/pkg/ucp/ucplog"
 	"github.com/radius-project/radius/pkg/version"
 	"oras.land/oras-go/v2/registry/remote"
 )
 
 const (
-	DevRecipesRegistry = "ghcr.io/radius-project"
+	// RecipeRepositoryPrefix is the prefix for the repository path.
+	RecipeRepositoryPrefix = "ghcr.io/radius-project/recipes/local-dev/"
 )
 
+type DevRecipe struct {
+	// NormalizedName is the normalized name of the recipe.
+	//
+	// For example, "mongodatabases".
+	NormalizedName string
+
+	// ResourceType is the resource type of the recipe.
+	//
+	// For example, "Applications.Datastores/mongoDatabases".
+	ResourceType string
+
+	// RepoPath is the repository path of the recipe.
+	//
+	// For example, "ghcr.io/radius-project/recipes/local-dev/mongodatabases".
+	RepoPath string
+}
+
+// NewDevRecipe creates a new DevRecipe object and returns it.
+func NewDevRecipe(normalizedName, resourceType, repoPath string) *DevRecipe {
+	return &DevRecipe{
+		NormalizedName: normalizedName,
+		ResourceType:   resourceType,
+		RepoPath:       repoPath,
+	}
+}
+
+// availableDevRecipes returns the list of available dev recipes.
+//
+// If we want to add a new recipe, we need to add it here.
```

Comment:
This isn't a great solution because it's hardcoded. Is this because it's not possible to list repositories in GHCR via ORAS?

I think it's urgent that we do **something** to fix the main branch since it's currently broken. If the solution is to hardcode the list of recipes, it's also urgent that we improve on that solution next.

Reply:
@ytimocin, could you please file an GH issue to track this improvement and reference it here?

Reply:
Since this PR is not addressing the issue https://github.com/radius-project/radius/issues/6506. You need to file an issue for adding devrecipes hard coded way, and use it as for this PR fix or you can just refer the release commit in the PR description.